### PR TITLE
[1LP][RFR] Automate: test_refresh_with_empty_iot_hub_azure

### PR DESCRIPTION
**Pull request by psimovec**
:heavy_check_mark:  requires https://github.com/ManageIQ/wrapanapi/pull/419
:stop_sign:  5.10.11 blocker found by test run https://bugzilla.redhat.com/show_bug.cgi?id=1756984 

{{ pytest: cfme/tests/cloud/test_providers.py::test_refresh_with_empty_iot_hub_azure -v --use-provider azure }}

### CODE REVIEW

File: cfme/tests/cloud/test_providers.py [hunk 1]
```diff
@@ -538,6 +539,42 @@ def test_azure_multiple_subscription(appliance, request, soft_assert):
...
+               create an IoT Hub in Azure (using free tier pricing is good enough):
+               $ az iot hub create --name rmanes-iothub --resource-group iot_rg
+        testSteps:
+            1. refresh azure provider
+        expectedResults:
+            1. no errors found in logs
+    Bugzilla:
+        1495318
+    """
+    result = LogValidator(
```
Comment (lina-nikiforova) [inline cfme/tests/cloud/test_providers.py]: This thing fits one line nicely :)
`result = LogValidator("/var/www/miq/vmdb/log/evm.log", failure_patterns=[r".*ERROR.*"])`

File: cfme/tests/cloud/test_providers.py [hunk 2]
```diff
@@ -538,6 +539,42 @@ def test_azure_multiple_subscription(appliance, request, soft_assert):
                                                                                      prov_b[0]))
 
 
+@pytest.mark.tier(1)
```
Comment (lina-nikiforova) [inline cfme/tests/cloud/test_providers.py]: should be 3, as 1 is the most important and this case has `caseimportance: low`

Comment (mshriver) [reply cfme/tests/cloud/test_providers.py]: Case Importance and tier are separate metadata.

Case importance is of low/medium/high/critical, and represents the how frequently the test case should be executed, along with how important it is when it fails.

Tier markers, integers 0, 1, 2, 3, equate to Case Level, which represents the test case layer from unit, function, system, to integration.

File: cfme/tests/cloud/test_providers.py [hunk 3]
```diff
@@ -538,6 +540,39 @@ def test_azure_multiple_subscription(appliance, request, soft_assert):
...
+        expectedResults:
+            1. no errors found in logs
+    Bugzilla:
+        1495318
+    """
+    result = LogValidator("/var/www/miq/vmdb/log/evm.log", failure_patterns=[r".*ERROR.*"])
+    result.start_monitoring()
+    azure = provider.mgmt
+    if not azure.has_iothub():
+        azure.create_iothub("potatoiothub")
```
Comment (mshriver) [inline cfme/tests/cloud/test_providers.py]: This needs to be randomized. If we're running multiple instances of the test framework concurrently, they could collide.

File: cfme/tests/cloud/test_providers.py [hunk 4]
```diff
@@ -538,6 +540,39 @@ def test_azure_multiple_subscription(appliance, request, soft_assert):
...
+        request.addfinalizer(lambda: azure.delete_iothub("potatoiothub"))
```
Comment (mshriver) [inline cfme/tests/cloud/test_providers.py]: :+1:

File: cfme/tests/cloud/test_providers.py (hunk 3 above)
Comment (lina-nikiforova) [reply cfme/tests/cloud/test_providers.py]: just a note: they could collide even in that case too. In our azure we can't have more than 1 iothub (1 is free).

**Pull request merged**